HowToRPC.md

```diff
@@ -0,0 +1,166 @@
+messages using gRPC’s HTTP/2-based messaging system, which supports request
+multiplexing, streaming APIs, and advanced flow control. To learn more about
+working with gRPC, visit [grpc.io/docs](https://grpc.io/docs).
+
+If gRPC support is unavailable, Google APIs can also be called using HTTP/1.1 or
+later using the fallback protocol described in the next section.
+
+### Authentication
+
+To use Google APIs, a client needs to authenticate with an API key or an OAuth
+token. For more information, see
+[the Google Cloud Authentication Overview](https://cloud.google.com/docs/authentication/).
+
+API keys can be obtained from the
+[Google Cloud Console > Credentials](http://console.cloud.google.com/apis/credentials)
+page. OAuth tokens can be obtained by [OAuth 2](https://oauth.net/2/) clients
+and libraries. For a sample command-line client, see the
+[oauth2l](https://github.com/google/oauth2l) on GitHub.
+
+## The gRPC Fallback Protocol
+
+Along with gRPC, most Google APIs support a simple fallback protocol that uses
+Protocol Buffers (protobuf) over HTTP. It allows clients to call Google APIs
+directly, often using standard library functions.
+
+This protocol uses fixed URLs to specify the RPC endpoints, and passes
+request/response messages as HTTP request/response body using HTTP POST. It uses
+normal HTTP headers to pass the RPC metadata, such as System Parameters.
+
+### URL
+
+RPC URLs have the following format:
+
+```
+URL ::= BaseUrl "/" Service "/" Method
+```
+
+- **BaseUrl.** This is the base URL published by service owners, either via
+  documentation or service discovery. For most Google APIs, the BaseUrl looks
+  like `https://language.googleapis.com/$rpc`. The base address can be found in
+  the API reference documentation where it is identified as the `Service name`.
+  For this example, `language.googleapis.com` is found in the Google Cloud
+  Natural Language API Reference.
+
+- **Service.** This is the fully qualified protobuf `service` name, such as
+  `google.cloud.language.v1.LanguageService`. In this case,
+  `google.cloud.language.v1` is the package name in
+  `google/cloud/language/v1/language_service.proto` and `LanguageService` is the
+  name of the service section found in this file.
+
+- **Method.** This is the protobuf `rpc` name, such as `AnnotateText`.
+
+### Requests
+
+RPC request messages are serialized and sent as the HTTP request body. For the
+server to handle the request properly, the client must set several HTTP request
+headers:
+
+- **Content-Type.** The request message format is specified by the Content-Type
+  header and must be `application/x-protobuf`.
+
+- **X-Goog-Api-Key.** This specifies a valid Google API key. It is optional if
+  the Authorization header is used.
+
+- **Authorization.** This specifies a valid Google OAuth access token in the
+  format of `Bearer {token}`. It is optional if the request is unauthenticated
+  or if an API key is used.
```

Comment:
unauthenticated and an API key is provided.

Reply:
I believe @lukesneeringer felt that "unauthenticated" would be unclear to some readers. 

I'll change this to "It is optional if an API key is provided."